testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,294 @@
+		},
+		"interval": "20s",
+		"records":  "1000",
+	}
+	cfg, err := ucfg.NewFrom(spigotConfig)
+	require.NoError(runner.T(), err)
+
+	logger, err := logrunner.New(cfg)
+	require.NoError(runner.T(), err)
+
+	go func() {
+		err := logger.Execute()
+		require.NoError(runner.T(), err)
+	}()
+
+	policyUUID := uuid.New().String()
+	unpr := false
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Unprivileged:   &unpr,
+	}
+
+	fixture, err := define.NewFixture(runner.T(), define.Version())
+	require.NoError(runner.T(), err)
+	runner.agentFixture = fixture
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+
+	basePolicy := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	// install system package
+	runner.InstallPackage(ctx, "system", "1.53.1", "agent_long_test_base_system_integ.json", uuid.New().String(), policyResp.ID)
+
+	// install cef
+	runner.InstallPackage(ctx, "cef", "2.16.2", "agent_long_test_cef.json", uuid.New().String(), policyResp.ID)
+
+}
+
+func (runner *ExtendedRunner) InstallPackage(ctx context.Context, name string, version string, cfgFile string, policyUUID string, policyID string) {
+	installPackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err := os.ReadFile(cfgFile)
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &installPackage)
+	require.NoError(runner.T(), err)
+
+	installPackage.Package.Version = version
+	installPackage.ID = policyUUID
+	installPackage.PolicyID = policyID
+	installPackage.Namespace = "default"
+	installPackage.Name = fmt.Sprintf("%s-long-test-%s", name, policyUUID)
+	installPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing %s package....", name)
+	_, err = runner.info.KibanaClient.InstallFleetPackage(ctx, installPackage)
+	require.NoError(runner.T(), err, "error creating fleet package")
+}
+
+func (runner *ExtendedRunner) TestHandleLeak() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "20m"
+	}
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(context.Background())
+
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			runner.T().Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
```

Comment:
If you ran this before the agent checked in and updated the component model, the system integration and cef integration inputs could be missing.

I think you need an assertion that the system/metrics, log, and cef inputs are actually in this list to avoid returning too early and sampling the wrong set of processes.

Reply:
There's also nothing here checking that these integrations actually do what they are supposed to do. So if we managed to configure them wrong and were measuring an agent doing nothing, we couldn't tell.